### PR TITLE
Make sure GZIPOutputStream is always closed

### PR DESCRIPTION
**Pull request by turbanoff**
Leave GZIPOutputStream open is dangerous as it leads to native memory leak.
Use try-finally to make sure stream is always closed

### CODE REVIEW

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java [hunk 1]
```diff
@@ -47,41 +48,42 @@ protected ByteArrayOutputStream initialValue()
...
             if ("/-/healthy".equals(contextPath)) {
                 osw.write(HEALTHY_RESPONSE);
             } else {
                 TextFormat.write004(osw,
                         registry.filteredMetricFamilySamples(parseQuery(query)));
             }
 
-            osw.flush();
             osw.close();
-            response.flush();
```
Comment (brian-brazil) [inline simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java:75]: Is it safe to remove all of these?

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java [hunk 2]
```diff
@@ -47,41 +48,42 @@ protected ByteArrayOutputStream initialValue()
      * Handles Metrics collections from the given registry.
      */
     static class HTTPMetricHandler implements HttpHandler {
-        private CollectorRegistry registry;
+        private final CollectorRegistry registry;
         private final LocalByteArray response = new LocalByteArray();
         private final static String HEALTHY_RESPONSE = "Exporter is Healthy.";
+        private final Charset UTF_8 = Charset.forName("UTF-8");
```
Comment (brian-brazil) [inline simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java:54]: This new indirection just makes the code harder to read for me.

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java (hunk 1 above)
Comment (turbanoff) [reply simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java:75]: 1. `flush` call before `close` is useless. According to javadoc - `close` should call `flush` first.
2. `ByteArrayOutputStream.flush` and `ByteArrayOutputStream.close` are no-op. So it's safe to remove them.

File: simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java (hunk 2 above)
Comment (brian-brazil) [reply simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java:54]: This is still here.

**Review by brian-brazil — COMMENTED** [review]
Can you also add the DCO?

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**